### PR TITLE
Julia Kozeeva Tasks 71-80

### PR DESCRIPTION
**Pull request by usersadvocate**

### CODE REVIEW

File: tasks/71_binary-encode/binary-encode.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+ * binaryEncode("fire")
+ * function binaryEncode returns an binary string of passed English sentance
+ * @param {String} str, string
+ * @return {String} binary string
+ */
+function binaryEncode(str) {
+  if (typeof(str) === 'undefined') {
+    return 'string';
+  }
+  const array = str.split('').map((el) => el.charCodeAt(0).toString(2) + '')
```
Comment (KalachevDev) [inline tasks/71_binary-encode/binary-encode.js]: `+ ''` is redundant here I think

File: tasks/72_carrying/carrying.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    if (typeof first !== 'number') {
+      return undefined;
+    } else {
+      return ((value) => {
+        if (typeof value !== 'number') {
+          return undefined;
+        } else {
+          return first + value;
+        }
+      });
```
Comment (KalachevDev) [inline tasks/72_carrying/carrying.js]: redundant brackets

File: tasks/73_check-for-palindromes/check-for-palindromes.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+ * //return true
+ * palindrome('eye')
+ * function palindrome checks if a string is a palindrome
+ * @param {String} str, input string
+ * @return {Boolean} a boolean
+ */
+function palindrome(str) {
+  str = str.toLowerCase().replace(/[^a-z0-9]/g, '');
+  const backword = str.split('').reverse().join('');
+  return backword === str ? true : false;
```
Comment (KalachevDev) [inline tasks/73_check-for-palindromes/check-for-palindromes.js]: `if (true) return true else return false`

File: tasks/74_dna-pairing/dna-pairing.js [hunk 4]
```diff
@@ -0,0 +1,23 @@
...
+ */
+function pairElement(str) {
+  const pairs = {
+    C: 'G',
+    G: 'C',
+    A: 'T',
+    T: 'A',
+  };
+  const dna = str.split('');
+  for (let i = 0; i < dna.length; i++) {
```
Comment (KalachevDev) [inline tasks/74_dna-pairing/dna-pairing.js]: Seem map is also suitable here

File: tasks/76_deep-object-clone/deep-object-clone.js [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+ * // return { a: 1, b: { c:10 } }
+ * pairElement({ a: 1, b: { c:10 } })
+ * function deepClone to copy an argument object
+ * @param {Object} obj, object for cloning
+ * @return {Object} copy original object
+ */
+function deepClone(obj) {
+  const clone = {};
+  for (let i in obj) {
+    if (obj[i] != null && typeof(obj[i])=='object') {
```
Comment (KalachevDev) [inline tasks/76_deep-object-clone/deep-object-clone.js]: What's about function?

File: tasks/80_sum-of-prime-indexes/sum-of-prime-indexes.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+ * @example
+ * // return a boolean
+ * isPrime(x)
+ * function isPrime checks prime numbers
+ * @param {Number} x, a number to check if it is prime
+ * @return {Boolean} true if number is prime, false if not
+ */
+function isPrime(x) {
+  for (let i = 2; i < x; i++) {
+    if (x % i === 0) {
```
Comment (KalachevDev) [inline tasks/80_sum-of-prime-indexes/sum-of-prime-indexes.js:11]: It's not necessary to go through all numbers square root is enough

File: tasks/71_binary-encode/binary-encode.js (hunk 1 above)
Comment (usersadvocate) [reply tasks/71_binary-encode/binary-encode.js]: yes, you're right

**Review by vovcyan — APPROVED** [review]

**Review by KalachevDev — APPROVED** [review]

**Pull request merged**